File: app/src/main/java/com/lttrung/dormitory/database/data/network/BillNetwork.java
package com.lttrung.dormitory.database.data.network;

import java.lang.reflect.Type;

import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Single;

@Singleton
public interface BillNetwork {
    Single<Type> fetchRoomBills();
    Single<Type> fetchElectricBills();
    Single<Type> fetchWaterBills();
}

File: app/src/main/java/com/lttrung/dormitory/database/data/network/BillNetwork.kt
package com.lttrung.dormitory.database.data.network

import com.lttrung.dormitory.database.data.network.models.ElectricBill
import com.lttrung.dormitory.database.data.network.models.WaterBill
import io.reactivex.rxjava3.core.Single
import java.lang.reflect.Type
import javax.inject.Singleton

@Singleton
interface BillNetwork {
    fun fetchRoomBills(): Single<Type>
    fun fetchElectricBills(): Single<List<ElectricBill>>
    fun fetchWaterBills(): Single<List<WaterBill>>
}